cni-plugin/main.go

```diff
@@ -0,0 +1,242 @@
+		logrus.SetLevel(logrus.DebugLevel)
+	} else if strings.EqualFold(logLevel, "info") {
+		logrus.SetLevel(logrus.InfoLevel)
+	} else {
+		// Default level
+		logrus.SetLevel(logrus.WarnLevel)
+	}
+
+	// Must log to Stderr because the CNI runtime uses Stdout as its state
+	logrus.SetOutput(os.Stderr)
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	logrus.Debugf("linkerd-cni: stdin to plugin: %v", string(stdin))
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("linkerd-cni: failed to parse network configuration: %v", err)
+	}
+
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("linkerd-cni: could not serialize prevResult: %v", err)
+		}
+
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("linkerd-cni: could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("linkerd-cni: could not convert result to current version: %v", err)
+		}
+		logrus.Debugf("linkerd-cni: prevResult: %v", conf.PrevResult)
+	}
+
+	return &conf, nil
+}
+
+// cmdAdd is called by the CNI runtime for ADD requests
+func cmdAdd(args *skel.CmdArgs) error {
+	logrus.Debug("linkerd-cni: cmdAdd, parsing config")
+	conf, err := parseConfig(args.StdinData)
+	if err != nil {
+		return err
+	}
+	configureLogging(conf.LogLevel)
+
+	if conf.PrevResult != nil {
+		logrus.WithFields(logrus.Fields{
+			"version":    conf.CNIVersion,
+			"prevResult": conf.PrevResult,
+		}).Debug("linkerd-cni: cmdAdd, config parsed")
+	} else {
+		logrus.WithFields(logrus.Fields{
+			"version": conf.CNIVersion,
+		}).Debug("linkerd-cni: cmdAdd, config parsed")
+	}
+
+	// Determine if running under k8s by checking the CNI args
+	k8sArgs := K8sArgs{}
+	args.Args = strings.Replace(args.Args, "K8S_POD_NAMESPACE", "K8sPodNamespace", 1)
+	args.Args = strings.Replace(args.Args, "K8S_POD_NAME", "K8sPodName", 1)
+	if err := types.LoadArgs(args.Args, &k8sArgs); err != nil {
+		return err
+	}
+
+	namespace := string(k8sArgs.K8sPodNamespace)
+	podName := string(k8sArgs.K8sPodName)
+	logEntry := logrus.WithFields(logrus.Fields{
+		"ContainerID": args.ContainerID,
+		"Pod":         podName,
+		"Namespace":   namespace,
+	})
+
+	if namespace != "" && podName != "" {
+		config, err := k8s.GetConfig(conf.Kubernetes.Kubeconfig, "linkerd-cni-context")
+		if err != nil {
+			return err
+		}
+
+		client, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			return err
+		}
+
+		pod, err := client.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		containsLinkerdProxy := false
+		for _, container := range pod.Spec.Containers {
+			if container.Name == "linkerd-proxy" {
```

Comment:
You can use constants from the "k8s" package here, instead of hardcoding the container name.

```suggestion
			if container.Name == k8s.ProxyContainerName {
```